test/utilities.js

```diff
@@ -1,6 +1,101 @@
 const { Capabilities, Builder } = require('selenium-webdriver');
 const { By, Key, until } = require('selenium-webdriver');
 
+
+const VAR_LANG_CLASS = 'variable-language';
+const COMMENT_STR_CLASS = 'cm-string';
+const COMMENT_CLASS = 'comment';
+const COMMENT_ATTR_CLASS = 'cm-attribute';
+const NO_CLASS = 'none';
+
+const textHighlightColors = {};
+textHighlightColors[VAR_LANG_CLASS] = 'rgb(127, 0, 85)';
+textHighlightColors[COMMENT_STR_CLASS] = 'rgb(0, 0, 255)';
+textHighlightColors[COMMENT_CLASS] = 'rgb(53, 125, 33)';
+textHighlightColors[COMMENT_ATTR_CLASS] = 'rgb(127, 0, 127)';
+textHighlightColors[NO_CLASS] = 'rgb(51, 51, 51)';
+
+const textColorClasses = Object.keys(textHighlightColors);
+
+// specify defaults
+const DEFAULT_SEARCH_FILTERS = {
+    owner: '*',
+    prefix: '*',
+    jobId: '*',
+    status: '*',
+};
+
+const compareFilters = (filter1, filter2) => {
+    const filterKeys = ['owner', 'prefix', 'jobId', 'status'];
+
+    let isMatch = true;
+    filterKeys.forEach(key => {
+        let key1 = '';
+        let key2 = '';
+
+        if (filter1[key]) {
+            key1 = filter1[key].toUpperCase();
+        }
+
+        if (filter2[key]) {
+            key2 = filter2[key].toUpperCase();
+        }
+
+        isMatch = isMatch && key1 === key2;
+    });
+    return isMatch;
+};
+
+const parseFilterText = filterText => {
+    const regex = /Owner= ?([^ ]*) ?Prefix= ?([^ ]*) ?JobId= ?([^ ]*) ?Status= ?([^ ]*) ?/;
+    const tokens = filterText.match(regex);
+    let owner;
+    let prefix;
+    let jobId;
+    let status;
+
+    tokens && tokens.forEach((val, idx) => {
+        switch (idx) {
+            case 1: owner = val; break;
+            case 2: prefix = val; break;
+            case 3: jobId = val; break;
+            case 4: status = val; break;
+            default: break;
+        }
+    });
+
+    return {
+        owner, prefix, jobId, status,
+    };
+};
+
+const parseJobText = text => {
+    const regex = /(.*):(.*) \[([^ ]*) ?(.*)\]/;
```

Comment:
There's a bug in the status render function in the UI code which means there's a chance that the white space between jobID and status is not present, I've raised an issue to resolve this but for the time being we can make that white space 0 or 1

```const regex = /(.*):(.*) ?\[([^ ]*) ?(.*)\]/;```

Reply:
This is the relevant issue to make sure the white space gets put in https://github.com/zowe/explorer-jes/issues/85

Reply:
taking both space or no space in account in modified regex